sygnal/apnspushkin.py

```diff
@@ -128,6 +136,20 @@ def __init__(self, name, sygnal, config):
         # without this, aioapns will retry every second forever.
         self.apns_client.pool.max_connection_attempts = 3
 
+    def _report_certificate_expiration(self, certfile):
+        """Export the epoch time that the certificate expires as a metric."""
+        with open(certfile, "rb") as f:
+            cert = f.read()
+
+        x509 = OpenSSL.crypto.load_certificate(OpenSSL.crypto.FILETYPE_PEM, cert)
+        # Convert from a string to a datetime object.
+        expiration_date = datetime.strptime(
+            x509.get_notAfter().decode(), "%Y%m%d%H%M%SZ"
+        )
+        # Convert the expiration time to seconds since the epoch.
+        epoch_time = int((expiration_date - datetime(1970, 1, 1)).total_seconds())
```

Comment:
/me stares at this a bit. does it do the the right thing with respect to timezones?

Also, can we not  just use `expiration_date.timestamp()`?

Reply:
oh, the incantation is possibly a workaround for `expiration_date` not having correct timezone info? I guess it would be helpful to write that as a comment.

Reply:
Ah, I grabbed this from another project which had Python 2.7 compatibility still. Didn't know `timestamp()` existed to be honest.